src/main/scala/org/scalacheck/ScalaCheckFramework.scala

```diff
@@ -99,13 +99,13 @@ private abstract class ScalaCheckRunner extends Runner {
       }
   }
 
-  def checkPropTask(taskDef: TaskDef, single: Boolean): BaseTask = new BaseTask(taskDef) {
+  def checkPropTask(taskDef: TaskDef, single: Boolean): BaseTask = new BaseTask(taskDef) { self =>
```

Comment:
Adding `self =>` seems to fix the compiler error in Dotty 0.24.

Now, I realize the other option is renaming parameter. That would produce signature compatibility issues, but fortunately, this class is `private`.